console/backend/pkg/model/converter.go

```diff
@@ -112,7 +125,40 @@ func GetTimeDiffer(startTime time.Time, endTime time.Time) (differ string) {
 		buffer.WriteString("m")
 		seconds = seconds - 60*minutes
 	}
-	buffer.WriteString(strconv.FormatFloat(seconds, 'g', -1, 64))
+	var s = int64(seconds)
+	buffer.WriteString(strconv.FormatInt(s, 10))
 	buffer.WriteString("s")
 	return buffer.String()
 }
+
+func ConvertDMONotebookToNotebookInfo(dmoNotebook *dmo.Notebook) NotebookInfo {
+	notebookInfo := NotebookInfo{
+		Id:             dmoNotebook.NotebookID,
+		Name:           dmoNotebook.Name,
+		NotebookStatus: dmoNotebook.Status,
+		Namespace:      dmoNotebook.Namespace,
+		Url:            dmoNotebook.Url,
+	}
+	if dmoNotebook.DeployRegion != nil {
+		notebookInfo.DeployRegion = *dmoNotebook.DeployRegion
+	}
+
+	if !dmoNotebook.GmtCreated.IsZero() {
+		notebookInfo.CreateTime = dmoNotebook.GmtCreated.Local().Format(JobInfoTimeFormat)
+	}
+	if !util.Time(dmoNotebook.GmtTerminated).IsZero() {
+		notebookInfo.EndTime = dmoNotebook.GmtTerminated.Local().Format(JobInfoTimeFormat)
+	}
+	now := time.Now()
+	if !dmoNotebook.GmtCreated.IsZero() {
+		if !util.Time(dmoNotebook.GmtTerminated).IsZero() {
+			// already terminated
+			notebookInfo.DurationTime = GetTimeDiffer(dmoNotebook.GmtCreated, *dmoNotebook.GmtTerminated)
+		} else {
+			notebookInfo.DurationTime = GetTimeDiffer(dmoNotebook.GmtCreated, now)
+		}
+	}
+	notebookInfo.UserName = *dmoNotebook.Owner
```

Comment:
`dmoNotebook.Owner` can be a nil value ?

Reply:
maybe, is this a problem ？

Reply:
then dereference `*dmoNotebook.Owner` may panic